### ISSUE TITLE
Support new .NET 5 lazy assembly loader

### ISSUE THREAD

**Issue by isc30** [isc30/blazor-lazy-loading#65]
Blazor team did a lot of work to introduce a partial implementation of the lib into blazor itself. This means that the `LazyAssemblyLoader` is a native part of blazor now! When .NET 5 RC1 is released, this library should shrink and reuse as much as it can from the official lazy loading support.

**Comment by knight1219** [comment]
Has there been any movement on this at all? I've been looking at this library as a way to help modularize a set of apps into Blazor and if it's using the native Blazor lazy loading that would be cool. This lib seems super easy to configure and use so I would love to start prototyping with it.

**Comment by isc30** [comment]
No movement cause R1 was super buggy. Now that they released R2 I'll give it another go and hopefully adapt the library to it :)

**Comment by isc30** [comment]
Had some time to look into this, some bits of the metadata generation are broken and need to investigate the .net5 internals to make it work. Will take some time.
If anyone wants to contribute i can push my changes and coordinate.

**Comment by Daeamon** [comment]
> If anyone wants
I'm want)

**Comment by ChristianWeyer** [comment]
Hi @isc30 - just a quick check whether you already know when you could have the .NET5-based new version available?
Thanks for the great lib!

**Comment by isc30** [comment]
Hi, I'm super busy lately but this is top in my priority list. I didn't
manage to make it work yet but im optimistic :)

>

**Comment by ChristianWeyer** [comment]
:-)

**Comment by ChristianWeyer** [comment]
Sorry for disturbing again. Maybe you have a remote idea of when the ETA of this update might be? 😇 @isc30

**Comment by isc30** [comment]
Hi, since stable .net5 was released some days ago, I'm working on making it work for that specific version.
I expect to have it by this week.

**Comment by isc30** [comment]
WIP in branch feature/net5
https://github.com/isc30/blazor-lazy-loading/compare/feature/net5

**Comment by isc30** [comment]
Blocked by the following error when trying to execute the custom MsBuild Task: GenerateManifest
![imagen](https://user-images.githubusercontent.com/10807051/99730039-1cc7b380-2abc-11eb-9dab-b92ea63e6a85.png)

any ideas if `net5.0` target dlls can contain msbuild Tasks?

**Comment by isc30** [comment]
Tried if porting only the components project to `.net5` would help with the msbuild task issue but nothing.
What I tried can be found in the `feature/net5.0-only-components` branch.

After some days of fighting with this I'm out of ideas on how to make a custom MsBuild task for in `.net5`.....

**Comment by ChristianWeyer** [comment]
Did you ask over at the .NET Core repo? They are quite helpful, usually.

**Comment by isc30** [comment]
No, I didn't reach them. I will be trying using `NestedTask` this afternoon and will reach them if it doesn't work.
If you want to help, the easiest way is to checkout `feature/net5.0-only-components`, build `BlazorLazyLoading.sln` and then build `demo/Demo.sln` to see the error. If we manage to sort this out, the rest of the implementation is quite straightforward

**Comment by isc30** [comment]
I'll be available in https://gitter.im/Blazor-Lazy-Loading/community
